perfmetrics/scripts/continuous_test/gke/common/utils.py

```diff
@@ -59,6 +60,236 @@ async def run_command_async(command_list, check=True, cwd=None):
+
+  await run_command_async(["sudo", "apt", "update", "-y"])
+
+  print("Checking for required tools...")
+  tools = {
+      "gcloud": ["gcloud", "--version"],
+      "git": ["git", "--version"],
+      "make": ["make", "--version"],
+      "kubectl": ["kubectl", "version", "--client=true"],
+      "gke-gcloud-auth-plugin": ["gke-gcloud-auth-plugin", "--version"],
+  }
+
+  for tool, version_cmd in tools.items():
+    try:
+      await run_command_async(version_cmd)
+    except (FileNotFoundError, subprocess.CalledProcessError):
+      if tool == "gcloud":
+        print("gcloud not found. Attempting to install...")
+        try:
+          await run_command_async(
+              ["sudo", "apt", "install", "-y", "google-cloud-sdk"]
+          )
+          # Re-check after installation
+          await run_command_async(version_cmd)
+        except (
+            FileNotFoundError,
+            subprocess.CalledProcessError,
+        ) as install_e:
+          print(
+              f"Error: Failed to install gcloud: {install_e}", file=sys.stderr
+          )
+          sys.exit(1)
+
+      if tool == "make":
+        print("make not found. Attempting to install...")
+        try:
+          await run_command_async(["sudo", "apt", "install", "-y", "make"])
+          await run_command_async(version_cmd)
+        except (
+            FileNotFoundError,
+            subprocess.CalledProcessError,
+        ) as install_e:
+          print(f"Error: Failed to install make: {install_e}", file=sys.stderr)
+          sys.exit(1)
+
+      if tool == "kubectl":
+        print("kubectl not found. Attempting to install...")
+        try:
+          await run_command_async(
+              ["sudo", "snap", "install", "kubectl", "--classic"]
+          )
+        except (FileNotFoundError, subprocess.CalledProcessError) as e:
+          print(f"Error: Failed to install kubectl: {e}", file=sys.stderr)
+          sys.exit(1)
+
+      elif tool == "gke-gcloud-auth-plugin":
+        print("gke-gcloud-auth-plugin not found. Attempting to install...")
+        try:
+          await run_command_async([
+              "sudo",
+              "apt",
+              "install",
+              "-y",
+              "google-cloud-sdk-gke-gcloud-auth-plugin",
+          ])
+        except (FileNotFoundError, subprocess.CalledProcessError) as e:
+          print(
+              f"Error: Failed to install gke-gcloud-auth-plugin: {e}",
+              file=sys.stderr,
+          )
+          sys.exit(1)
+
+      else:
+        print(
+            f"Error: Required tool '{tool}' is not installed. Please install it"
+            " before running.",
+            file=sys.stderr,
+        )
+        sys.exit(1)
+  print("All required tools are installed.")
+
+
+async def setup_gke_cluster(
+    project_id,
+    zone,
+    cluster_name,
+    network_name,
+    subnet_name,
+    region,
+    machine_type,
+    node_pool_name,
+    default_mtu,
```

Comment:
Why do we need "default_mtu" as a parameter? Are we using any value other than the DEFAULT_MTU. If not, then let's declare the constant in this file and use that for creating the network.